torch/_torch_docs.py

```diff
@@ -5390,9 +5392,9 @@ def merge_dicts(*dicts):
 
 add_docstr(torch.argmin,
            r"""
-argmin(input) -> LongTensor
+argmin(input, dim=None, keepdim=False) -> LongTensor
 
-Returns the indices of the minimum value of all elements in the :attr:`input` tensor.
+Returns the indices of the minimum values of a tensor.
```

Comment:
I don't think this updated statement is true.

```
torch.argmin(torch.tensor((1., 1.)))
: tensor(0)
```

If dim is None then argmin computes the flattened index of a minimal element in input. If dim is specified then I have a hard time explaining what argmin does. It's like... it computes the indices of minimal elements of... something? I don't think we can great language for this.

Looking through our internal code bases there are very few (if any) uses of this function with dim or axis (for np.argmin) set, which doesn't surprise me. You can't even use the results of this function as indices to get the minima!

```
t = torch.randn(2, 2, 2)
t
: tensor(
  [[[ 0.5317, -0.2113],
    [ 1.0522, -1.1861]],
   [[-1.8205, -0.7039],
    [-0.3392,  0.2466]]])

minima_indices = torch.argmin(t, dim=1)
minima_indices
: tensor(
  [[0, 1],
   [0, 0]])

t[minima_indices]
: 
tensor(
  [[[[ 0.5317, -0.2113],
     [ 0.5317,  0.5317]],
    [[ 1.0522, -1.1861],
     [ 1.0522,  1.0522]]],
   [[[-1.8205, -0.7039],
     [-1.8205, -1.8205]],
    [[-0.3392,  0.2466],
     [-0.3392, -0.3392]]]])
```

Reply:
> The NumPy documentation says "along an axis," which I guess is OK? I wonder if "along an axis" is ever formally defined?

So change to _"Returns the indices of the minimum values along a dimension"_? I'm not sure there's a formal definition of "along an axis" anywhere, it's usually made clear by examples.



> The NumPy documentation correctly pairs this function with take_along_axis, which PyTorch doesn't have (but should!).

Agreed. Tracked already in gh-38349

Reply:
That seems OK but we should be clear that it returns the index of *a* minimal value. That is, if a tensor has multiple minimal values then this could return the flattened index of *any* of them, and will still only return one flattened index.

Reply:
So like... "Computes the flattened index of a minimal value of input when dim is None, or the indices of a minimal value along each dimension when dim is specified"?

Something about that sentence still seems a little off. It's hard to articulate that when dim is specified then multiple minimal values are returned but still just one "along each axis" is picked.

Reply:
Note that there's already a prominent note right below this sentence about duplicates:

<img width="502" alt="image" src="https://user-images.githubusercontent.com/98330/102817962-ffdf1280-43d0-11eb-9ce7-0cdddad76902.png">

The first sentence of a docstring shouldn't be too long, since it goes into the autosummary table. Capturing corner cases in that first sentence isn't necessary. So `the value` -> `a value` should do it.